cristin-person/src/main/java/no/unit/nva/cristin/person/update/UpdateCristinPersonHandler.java

```diff
@@ -58,25 +58,24 @@ protected Void processInput(String input, RequestInfo requestInfo, Context conte
 
         logger.info(LOG_IDENTIFIERS, extractCristinIdentifier(requestInfo), extractOrgIdentifier(requestInfo));
 
-        ObjectNode objectNode = readJsonFromInput(input);
-        String personId = getValidPersonId(requestInfo);
-
         if (clientCanUpdateAllFields(requestInfo)) {
+            var objectNode = readJsonFromInput(input);
             PersonPatchValidator.validate(objectNode);
-            ObjectNode cristinJson = new CristinPersonPatchJsonCreator(objectNode).create().getOutput();
+            var cristinJson = new CristinPersonPatchJsonCreator(objectNode).create().getOutput();
             checkHasFields(cristinJson);
 
             if (cristinJson.has(CRISTIN_EMPLOYMENTS)) {
-                return apiClient.updatePersonInCristin(personId, cristinJson,
-                                                       extractCristinInstitutionIdentifier(requestInfo));
+                return apiClient.updatePersonInCristin(
+                    getValidPersonId(requestInfo), cristinJson, extractCristinInstitutionIdentifier(requestInfo));
             } else {
-                return apiClient.updatePersonInCristin(personId, cristinJson);
+                return apiClient.updatePersonInCristin(getValidPersonId(requestInfo), cristinJson);
             }
         } else if (clientCanUpdateOwnData(requestInfo)) {
+            var objectNode = readJsonFromInput(input);
             String personIdFromCognito = parseLastPartOfPersonCristinIdFromCognito(requestInfo).orElseThrow();
-            checkIdentifiersMatch(personId, personIdFromCognito);
+            checkIdentifiersMatch(getValidPersonId(requestInfo), personIdFromCognito);
```

Comment:
you are calling getValidPersonId 3 times, is it not better to use a local variable? or is it better to avoid local variables?

Reply:
and readJsonFromInput is called twice